### ISSUE TITLE
[Bug]: DN release datasyncservice stucked

### ISSUE THREAD

**Issue by XuanYang-cn** [milvus-io/milvus#12187]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

may cause 1200% CPU (not verified yet)

**Comment by XuanYang-cn** [comment]
Possibly fixed by #12260

**Comment by XuanYang-cn** [comment]
Not reproducing anymore.

Reopen if reproducing.
